### DISCUSSION TITLE
ERROR! the role 'metatrader' was not found in ansible.legacy

### DISCUSSION THREAD

**Discussion by NirobNabil** [EA31337/ansible-role-mt-runner#4] [General]
I opened a discussion on your EA-Tester repository about running backtesting on mt5 where you redirected me to here.
 
I was able to set up ansible-mt-runner which installed metatrader ( Even though there were some problem. I've created issues based on these ). But i think the current master branch doesn't contain configuration details for running backtests out of the box. So i switched to dev branch but now i get this error that says `ERROR! the role 'metatrader' was not found in ansible.legacy`

Would it be possible for you to create a dummy project with working backtests that i can run and then work on it incrementally. Or it would be great if you can atleast guide me on what steps i need to follow to get it working.

Thank you

**Comment by kenorb** [comment]
The `mt-runner` role depends on `metatrader` role. Dependencies are defined in `meta/main.yml` (including role name) and I think they should be automatically picked by Ansible. If not, install it via `ansible-galaxy`, e.g.

    ansible-galaxy install git+https://github.com/EA31337/ansible-role-mt-runner.git

This will automatically install roles with the right names.

**Comment by kenorb** [comment]
There were several recent fixes in `dev` branch, so after installing this role via:

    ansible-galaxy install git+https://github.com/EA31337/ansible-role-mt-runner.git,dev

After install, you should see something like:

```
Starting galaxy role install process
- extracting ansible-role-mt-runner to /root/.ansible/roles/ansible-role-mt-runner
- ansible-role-mt-runner (dev) was installed successfully
- adding dependency: ea31337.metatrader (dev)
- extracting ea31337.metatrader to /root/.ansible/roles/ea31337.metatrader
- ea31337.metatrader (dev) was installed successfully
- adding dependency: ea31337.wine (dev)
- adding dependency: ea31337.xvfb (dev)
- extracting ea31337.wine to /root/.ansible/roles/ea31337.wine
- ea31337.wine (dev) was installed successfully
- extracting ea31337.xvfb to /root/.ansible/roles/ea31337.xvfb
- ea31337.xvfb (dev) was installed successfully
# /root/.ansible/roles
- ansible-role-mt-runner, dev
- ea31337.xvfb, dev
- ea31337.wine, dev
```

To create a playbook, check the examples from `molecule/default/converge.yml` or `molecule/mt5-ea/converge.yml`.

For the example, check the following Colab: <https://colab.research.google.com/drive/1ybsqnVANuiJBdnza47e9VRIYFK4E1xKZ>.
